plugins/policy-recommendation/policy_recommendation_job.py

```diff
@@ -0,0 +1,849 @@
+        rm_labels: Remove automatically generated Pod labels including 'pod-template-hash', 'controller-revision-hash', 'pod-template-generation'.
+        to_services: Use the toServices feature in ANP, only works when option is 1 or 2.
+
+    Returns:
+        A list of recommended policies, each recommended policy is a string of YAML format.
+    """
+    recommend_policies = []
+    sql_query = generate_sql_query(table_name, limit, start_time, end_time, True)
+    unprotected_flows_df = read_flow_df(db_jdbc_address, sql_query, rm_labels)
+    recommend_policies += recommend_policies_for_unprotected_flows(unprotected_flows_df, option, to_services)
+    if option in [1, 2]:
+        sql_query = generate_sql_query(table_name, limit, start_time, end_time, False)
+        trusted_denied_flows_df = read_flow_df(db_jdbc_address, sql_query, rm_labels)
+        recommend_policies += recommend_policies_for_trusted_denied_flows(trusted_denied_flows_df, to_services)
+    return recommend_policies
+
+def main(argv):
+    db_jdbc_address = "jdbc:clickhouse://clickhouse-clickhouse.flow-visibility.svc:8123"
+    flow_table_name = "default.flows"
+    result_table_name = "default.recommendations"
+    recommendation_type = 'initial'
+    limit = 0
+    option = 1
+    start_time = ""
+    end_time = ""
+    ns_allow_list = NAMESPACE_ALLOW_LIST
+    id = ""
+    rm_labels = True
+    to_services = True
+    help_message = """
+    Start the policy recommendation spark job.
+
+    Options:
+    -h, --help: Show help message.
+    -t, --type=initial: {initial|subsequent} Indicates this recommendation is an initial recommendion or a subsequent recommendation job.
+    -d, --db_jdbc_url=None: The JDBC URL used by Spark jobs connect to the ClickHouse database for reading flow records and writing results.
+        jdbc:clickhouse://clickhouse-clickhouse.flow-visibility.svc:8123 is the ClickHouse JDBC URL used by default.
+    -l, --limit=0: The limit on the number of flow records read from the database. 0 means no limit.
+    -o, --option=1: Option of network isolation preference in policy recommendation.
+        Currently we have 3 options:
+        1: Recommending allow ANP/ACNP policies, with default deny rules only on applied to Pod labels which have allow rules recommended.
+        2: Recommending allow ANP/ACNP policies, with default deny rules for whole cluster.
+        3: Recommending allow K8s network policies, with no deny rules at all.
+    -s, --start_time=None: The start time of the flow records considered for the policy recommendation. 
+        Format is YYYY-MM-DD hh:mm:ss in UTC timezone. Default value is None, which means no limit of the start time of flow records.
+    -e, --end_time=None: The end time of the flow records considered for the policy recommendation.
+        Format is YYYY-MM-DD hh:mm:ss in UTC timezone. Default value is None, which means no limit of the end time of flow records.
```

Comment:
Since we may do not have proper flow records in DB, maybe we need a warn message to indicate this case?